circuits/web/tools.py

```diff
@@ -449,3 +451,20 @@ def gzip(response, level=4, mime_types=("text/html", "text/plain",)):
     return httperror(
         response.request, response, 406, description="identity, gzip"
     )
+
+
+class BehindProxy(BaseComponent):
```

Comment:
``ProxyFilter``, ``RealIP``, ``ProxyIP``? If ``BehindProxy`` makes more sense though leave it :)